smarts/env/wrappers/metrics.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2022. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import copy
+import functools
+from dataclasses import dataclass, fields
+from typing import Any, Dict, Set, TypeVar
+
+import gym
+
+from smarts.env.wrappers.metric import termination
+from smarts.env.wrappers.metric.costs import CostFuncs, Costs
+from smarts.env.wrappers.metric.counts import Counts
+
+_MAX_STEPS = 800
```

Comment:
Now, the max episode steps is obtained directly from each agent spec's `max_episode_steps` attribute. This will be used to normalise the number of steps taken by each agent in each episode.

Reply:
Now, the max episode steps is obtained directly from each agent spec's `max_episode_steps` attribute. The `_MAX_STEPS` constant has been removed.